ably/realtime_conn.go

```diff
@@ -40,7 +41,14 @@ type Conn struct {
 	// reconnecting tracks if we have issued a reconnection request. If we receive any message
 	// with this set to true then its the first message/response after issuing the
 	// reconnection request.
-	reconnecting bool
+	reconnecting atomic.Value
```

Comment:
I'm not sure we should introduce another lock when we already have one to cover the data accessed by both the connection event loop and calls from the outside, namely `c.state`. This is the lock that we hold on `eventloop` below for data we access when when a message arrives, which is also the case here.

Reply:
Done at 909ef4b07af24f08e2151b7866e6d7342af9a948. But really, we should revisit locking here, we're unlocking and locking several times in the same sequence which logically is atomic.